File: test-scaffolds/csharp/ExampleTest.cs
// NUnit scaffold. Class name must match file name and end in Tests.
// For xUnit replace [TestFixture]/[Test] with no attribute / [Fact].
// Replace MyService and placeholder values with real names.

using NUnit.Framework;
using Moq;

namespace MyProject.Tests;

[TestFixture]
public class ExampleTest
{
    private Mock<IDependency> _mockDep = null!;
    private MyService _svc = null!;

    [SetUp]
    public void SetUp()
    {
        _mockDep = new Mock<IDependency>();
        _svc = new MyService(_mockDep.Object);
    }

    [Test]
    public void DoSomething_ValidInput_ReturnsExpectedResult()
    {
        _mockDep.Setup(d => d.Fetch("key")).Returns("value");
        var result = _svc.DoSomething("key");
        Assert.That(result, Is.EqualTo("expected_output"));
    }

    [Test]
    public void DoSomething_NullInput_ThrowsArgumentNullException()
    {
        Assert.That(() => _svc.DoSomething(null!), Throws.ArgumentNullException);
    }

    [Test]
    public async Task DoSomethingAsync_ValidInput_ReturnsExpectedResult()
    {
        _mockDep.Setup(d => d.FetchAsync("key")).ReturnsAsync("async_value");
        var result = await _svc.DoSomethingAsync("key");
        Assert.That(result, Is.EqualTo("async_expected_output"));
    }
}

File: test-scaffolds/php/ExampleTest.php
<?php

declare(strict_types=1);

// PHPUnit scaffold. Class name must match file name and extend TestCase.
// Replace MyService and placeholder values with real names.
// Run: ./vendor/bin/phpunit tests/ExampleTest.php

namespace App\Tests;

use App\MyService;
use App\Contracts\DependencyInterface;
use PHPUnit\Framework\TestCase;
use PHPUnit\Framework\MockObject\MockObject;

final class ExampleTest extends TestCase
{
    private MockObject $mockDep;
    private MyService $svc;

    protected function setUp(): void
    {
        $this->mockDep = $this->createMock(DependencyInterface::class);
        $this->svc = new MyService($this->mockDep);
    }

    public function testDoSomethingReturnsExpectedValueForValidInput(): void
    {
        $this->mockDep->method('fetch')->with('key')->willReturn('value');
        $result = $this->svc->doSomething('key');
        $this->assertEquals('expected_output', $result);
    }

    public function testDoSomethingThrowsForEmptyInput(): void
    {
        $this->expectException(\InvalidArgumentException::class);
        $this->expectExceptionMessage('must not be empty');
        $this->svc->doSomething('');
    }
}
